container.go

```diff
@@ -482,6 +482,113 @@ func (c *Client) TopContainer(id string, psArgs string) (TopResult, error) {
+	Stats           DetailedMemoryStats `json:"stats"`
+	MaxUsage        uint64              `json:"max_usage"`
+	Usage           uint64              `json:"usage"`
+	Failcnt         uint64              `json:"failcnt"`
+	Limit           uint64              `json:"limit"`
+	PercentageInUse float64
+}
+
+// DetailedMemoryStats describes detailed data related to the container's memory usage.
+type DetailedMemoryStats struct {
+	TotalPgmajfault         uint64 `json:"total_pgmajfault"`
+	Cache                   uint64 `json:"cache"`
+	MappedFile              uint64 `json:"mapped_file"`
+	TotalInactiveFile       uint64 `json:"total_inactive_file""`
+	Pgpgout                 uint64 `json:"pgpgout"`
+	Rss                     uint64 `json:"rss"`
+	TotalMappedFile         uint64 `json:"total_mapped_file"`
+	Writeback               uint64 `json:"writeback"`
+	Unevictable             uint64 `json:"unevictable"`
+	Pgpgin                  uint64 `json:"pgpgin"`
+	TotalUnevictable        uint64 `json:"total_unevictable"`
+	Pgmajfault              uint64 `json:"pgmajfault"`
+	TotalRss                uint64 `json:"total_rss"`
+	TotalRssHuge            uint64 `json:"total_rss_huge"`
+	TotalWriteback          uint64 `json:"total_writeback"`
+	TotalInactiveAnon       uint64 `json:"total_inactive_anon"`
+	RssHuge                 uint64 `json:"rss_huge"`
+	HierarchicalMemoryLimit uint64 `json:"hierarchical_memory_limit"`
+	TotalPgfault            uint64 `json:"total_pgfault"`
+	TotalActiveFile         uint64 `json:"total_active_file"`
+	TotalPgpgout            uint64 `json:"total_pgpgout"`
+	TotalCache              uint64 `json:"total_cache"`
+	InactiveAnon            uint64 `json:"inactive_anon"`
+	ActiveFile              uint64 `json:"active_file"`
+	Pgfault                 uint64 `json:"pgfault"`
+	InactiveFile            uint64 `json:"inactive_file"`
+	TotalPgpgin             uint64 `json:"pgpgin"`
+}
+
+// CPUStats describes the data related to the container's CPU usage.
+type CPUStats struct {
+	CPUUsage        CPUUsage `json:"cpu_usage"`
+	SystemCPUUsage  uint64   `json:"system_cpu_usage"`
+	PercentageInUse float64
+}
+
+// CPUUsage describes the data related to the container's CPU usage.
+type CPUUsage struct {
+	PercpuUsage       []uint64 `json:"percpu_usage"`
+	UsageInUsermode   uint64   `json:"usage_in_usermode"`
+	TotalUsage        uint64   `json:"total_usage"`
+	UsageInKernelmode uint64   `json:"usage_in_kernelmode"`
+}
+
+// ContainerStats describes the data as it is retrieved from the /stats/ endpoint.
+type ContainerStats struct {
+	Read    string       `json:"read"`
+	Stream  string       `json:"stream,omitempty"`
+	Network NetworkStats `json:"network"`
+	Memory  MemoryStats  `json:"memory_stats"`
+	CPU     CPUStats     `json:"cpu_stats"`
+	EOF     string
+}
+
+// uses a different struct to unmarshal to when we're calling the stats endpoint
+func decodeStats(r io.Reader, stats chan ContainerStats) error {
+	dec := json.NewDecoder(r)
```

Comment:
That's just a question, should we add a ``defer close(stats)`` here, to signal readers that after this function returns, no new messages will be sent in the channel?